bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/template/factory/impl/graces/Graces.scala

```diff
@@ -1,14 +1,69 @@
 package edu.gemini.phase2.template.factory.impl.graces
 
 import edu.gemini.phase2.template.factory.impl._
+import edu.gemini.shared.skyobject.Magnitude.Band
 import edu.gemini.spModel.gemini.graces.blueprint.SpGracesBlueprint
-import edu.gemini.pot.sp.ISPGroup
+import edu.gemini.pot.sp.{SPComponentType, ISPGroup}
+import edu.gemini.spModel.gemini.visitor.VisitorInstrument
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.shared.util.immutable.ScalaConverters._
 
-case class Graces(blueprint: SpGracesBlueprint) extends GroupInitializer[SpGracesBlueprint] {
+import SpGracesBlueprint.ReadMode._
+import SpGracesBlueprint.FiberMode._
+
+case class Graces(blueprint: SpGracesBlueprint, exampleTarget: Option[SPTarget]) extends GroupInitializer[SpGracesBlueprint] with TemplateDsl2[VisitorInstrument] {
   val program = "GRACES PHASE I/II MAPPING BPS"
-  val targetGroup = Seq(1, 2)
-  val baselineFolder = Seq.empty
-  val notes = Seq("N FIBERS")
 
-  def initialize(group:ISPGroup, db:TemplateDb): Maybe[Unit] = Right(())
+  def instCompType: SPComponentType =
+    VisitorInstrument.SP_TYPE
+
+  def seqConfigCompType: SPComponentType =
+    sys.error("No sequence component type for visitor instrument GRACES")
+
+  // This seems to be necessary, sorry.
+  var db:Option[TemplateDb] = None
+  override def initialize(db:TemplateDb):Maybe[ISPGroup] =
```

Comment:
It's defined as `Either[String,A]` in the template code. Not a great idea in hindsight but I want to be consistent with the other stuff in that package.